tests/accessor/generic_accessor_linearization.h

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides common code for generic sycl::accessor correct linearization test
+//  For multidimensional accessors the iterator linearizes the data according
+//  to Section 3.11.1
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_GENERIC_ACCESSOR_LINEAR_COMMON_H
+#define SYCL_CTS_GENERIC_ACCESSOR_LINEAR_COMMON_H
+#include "accessor_common.h"
+
+namespace generic_accessor_linearization {
+using namespace accessor_tests_common;
+
+template <typename T, typename AccessT, typename DimensionT, typename TargetT>
+class run_linearization_tests {
+  static constexpr sycl::access_mode AccessMode = AccessT::value;
+  static constexpr int dims = DimensionT::value;
+  static constexpr sycl::target Target = TargetT::value;
+  using AccT = sycl::accessor<T, dims, AccessMode, Target>;
+
+ public:
+  void operator()(const std::string &type_name,
+                  const std::string &access_mode_name,
+                  const std::string &target_name) {
+    auto queue = util::get_cts_object::queue();
+    auto r = util::get_cts_object::range<dims>::get(1, 1, 1);
+
+    SECTION(
+        get_section_name<dims>(type_name, access_mode_name, target_name, "")) {
+      constexpr size_t range_size = 8;
+      constexpr size_t buff_size = (dims == 3) ? 8 * 8 * 8 : 8 * 8;
```

Comment:
Use prime numbers everywhere to detect more errors.

Reply:
Changed to 2

Reply:
I meant prime numbers between each others, like 2, 3 and 5 in the case of 3D, to lower the probability to have the use of wrong axis being undetected.